changelog.d/inconsistent-indentation

```diff
@@ -0,0 +1,12 @@
+synopsis: Warn about inconsistent indentation
+packages: Cabal-syntax
+prs: #8975
+
+description:
+    Make readFields warn about inconsistent indentation.
```

Comment:
Could the changelog start in a more approachable manner explaining the overall effect of the change for a user (e.g. "if you have this and that in your cabal file, you'll be warned now" or some such...)? Starting referencing API from get go feels intimidating. Changelogs go to release notes and as such will probably be read by many people, most of whom are unfamiliar with Cabal code base and API.

Reply:
> > I had to adjust lexer slightly as BOM and NBSP where throwing source locations a bit off causing cascading indentation warnings.
> 
> This sounds a bit worrying. Have you seen the discussion at #8896? Don't we step on each others toes here?
> 
> At the very least, the Lexer change sounds like something that should go into a separate commit.

I haven't. Can you summarize?

I split that change into separate commit.

Also I modified the changelog entry. Please suggest concrete edits if you want it to be changed further.

Reply:
Thank you.

@hsyl20, as the last person to touch Lexer.x, could you perhaps share your view on the proposed change and say if it looks reasonable to you?

Reply:
Thank you!